src/MongoDB.Driver.Core/Core/WireProtocol/Messages/CompressedMessage.cs

```diff
@@ -0,0 +1,83 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using MongoDB.Bson.IO;
+using MongoDB.Driver.Core.Compression;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.Core.WireProtocol.Messages.Encoders;
+
+namespace MongoDB.Driver.Core.WireProtocol.Messages
+{
+    /// <summary>
+    /// Represents a compressed message.
+    /// </summary>
+    public class CompressedMessage : MongoDBMessage
+    {
+        private readonly CompressorType _compressorType;
+        private readonly MongoDBMessage _originalMessage;
+        private readonly BsonStream _originalMessageStream;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CompressedMessage"/> class.
+        /// This constructor is used when reading a compressed message.
+        /// </summary>
+        /// <param name="originalMessage">The original message.</param>
+        /// <param name="compressorType">The compressor type.</param>
+        public CompressedMessage(MongoDBMessage originalMessage, CompressorType compressorType) : this(originalMessage, null, compressorType)
```

Comment:
It's better to put calls to base/other constructors on the next line:

    : this(originalMessage, null, compressorType)

it's too easy to not see it when it's at the end of the line.

Reply:
If we're just going to call the other constructor from this one perhaps we should only have a single constructor after all and the ONE place that calls this constructor can pass null for the originalMessageStream.